API/Controllers/ClientsController.cs

```diff
@@ -0,0 +1,44 @@
+using API.DTOs;
+using API.Entities;
+using API.Services.Interfaces;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[EnableCors]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class ClientsController(IOAuthClientService clientService) : Controller
+{
+    [HttpPatch("{id:int}/ratelimit")]
+    [Authorize(Roles = "admin")]
+    [EndpointSummary("Patches the ratelimit for a given client")]
+    public async Task<Results<BadRequest, NotFound<string>, Ok<OAuthClientDTO>>> PatchRatelimitAsync(int id, [FromBody] JsonPatchDocument<RateLimitOverrides> patchedOverrides)
+    {
+        OAuthClientDTO? currentClient = await clientService.GetAsync(id);
+
+        if (currentClient == null)
+        {
+            return TypedResults.NotFound("Client not found");
```

Comment:
```suggestion
            return TypedResults.NotFound();
```

"Client not found" is inferable by just the error itself. Like I mentioned to Convex, I think we should reserve using custom error messages only for when we need to communicate relevant information that cannot be inferred. Make sure return type matches.

Reply:
Ah right, yep sounds good.